compiler-rt/lib/ctx_profile/CtxInstrProfiling.cpp

```diff
@@ -0,0 +1,38 @@
+//===- CtxInstrProfiling.cpp - contextual instrumented PGO000000 ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CtxInstrProfiling.h"
+#include "sanitizer_common/sanitizer_allocator_internal.h"
+#include "sanitizer_common/sanitizer_common.h"
+#include "sanitizer_common/sanitizer_dense_map.h"
+#include "sanitizer_common/sanitizer_mutex.h"
+#include "sanitizer_common/sanitizer_placement_new.h"
+#include "sanitizer_common/sanitizer_thread_safety.h"
+
+#include <assert.h>
+
+using namespace __ctx_profile;
+
+Arena *Arena::allocateNewArena(size_t Size, Arena *Prev) {
```

Comment:
Is this for testing? I assume you are going to link sanitizer_common into production binaries?

Reply:
It's part of the design. Yes, when collecting profiles, sanitizer_common would be linked.

Reply:
If so, and you are using just InternalAlloc, I'd recommed to switch to malloc() or mmap().
At leaset in google we never link sanitizer_common in prod, only preprod.
Also it will reduce future linking issues if we need to test this with ubsan, or other sanitizer together,

Reply:
OK, but I'll also need other things from sanitizer_common, like __sanitizer::Vector. This is only being used for instrumented profile collection, it's similar to `profile`  or `memprof`.

(that being said, I can switch to malloc or mmap here, though, probably more efficient)

Reply:
Discussed offline - if sanitizers and production compatibility is not goal right now we can re-use code, and re-implement/refactor stuff later, when needed.